internal/sync2/sleep_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information
+
+package sync2_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"storj.io/storj/internal/sync2"
+)
+
+func TestSleep(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	start := time.Now()
+	if !sync2.Sleep(ctx, time.Second) {
+		t.Error("expected true as result")
+	}
+	if time.Since(start) < time.Second/2 {
```

Comment:
Why checking if the sleep took 500 ms instead of 1 sec? Aren't we expecting a full second of sleep?

Reply:
Sleep is somewhat annoying that it can sleep less than a second as well. It has an +- error in terms of system tick granularity.

Reply:
added a bit clarity to it